### PR TITLE
Changelog

### PR DESCRIPTION
**Pull request by SwartzCr**
adds a changelog file

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+their certs while experimenting with the client, in some cases 
+hitting issuance rate limits 
+* numerous Apache configuration parser fixes 
+* avoid attempting to issue for unqualified domain names like 
+"localhost" 
+* fix --webroot permission handling for non-root users 
+
+More details about these changes can be found on our GitHub repo:
+https://github.com/letsencrypt/letsencrypt/issues?q=milestone%3A0.1.1 
+
```
Comment (bmw) [inline CHANGELOG.md]: nit: We can probably delete some/all of these blank lines.

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+* A non-interactive mode which can be enabled by including `-n` or 
+`--non-interactive` on the command line. This can be used to 
+guarantee the client will not prompt when run automatically using 
+cron/systemd. 
+* Preparation for the new letsencrypt-auto script. Over the past 
+couple months, we've been working on increasing the reliability and 
+security of letsencrypt-auto. A number of changes landed in this 
+release to prepare for the new version of this script. 
+
+
```
Comment (bmw) [inline CHANGELOG.md]: nit: there's an extra space here that's not present for other entries

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+successful, failed, or not yet due. 
+* A `--dry-run` flag has been added to help with testing configuration 
+without affecting production rate limits. Currently supported by the 
+`renew` and `certonly` subcommands, providing `--dry-run` on the command 
+line will obtain certificates from the staging server without saving the 
+resulting certificates to disk. 
+* Major improvements have been added to letsencrypt-auto. This script 
+has been rewritten to include full support for Python 2.6, the ability 
+for letsencrypt-auto to update itself, and improvements to the 
+    stability, security, and performance of the script.
```
Comment (bmw) [inline CHANGELOG.md]: nit: weird spacing

File: CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+* A patch fixing problems of using letsencrypt renew with configuration files
+from private beta has been added. 
+
+More details about these changes can be found on our GitHub repo:
+https://github.com/letsencrypt/letsencrypt/issues?q=is%3Aissue+milestone%3A0.4.2 
+
+# 0.4.1
+## 02/29/2016
+
+* Apache parsing errors with some configurations
```
Comment (bmw) [inline CHANGELOG.md]: This list doesn't really make sense out of context. In the original e-mail, I had the following text above these bullet points:

> This is a bug fix release which includes patches for the following
problems:

Listing those problems on their own is a bit awkward. I'd prepend a word like "fixes", "resolves", or whatever other word your favorite thesaurus comes up with so these phrases make more sense on their own.

File: CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+## 05/12/2016 
+
+* Renamed the client from `letsencrypt` to `certbot`
+* Fixed a small json deserialization error
+* Versioned the datetime dependency in setup.py
+* Preserve domain order in generated CSRs
+* Some minor bug fixes
+
+More details about these changes can be found on our GitHub repo:
+https://github.com/certbot/certbot/issues?utf8=%E2%9C%93&q=is%3Aissue%20milestone%3A0.6.0%20is%3Aclosed%20
```
Comment (bmw) [inline CHANGELOG.md]: nit: In the other links Peter or I removed `utf8=%E2%9C%93` from the query string. Let GH determine whether or not serve content as UTF-8.

File: CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,331 @@
...
+custom CSR 
+* a number of bug fixes 
+
+More details about these changes can be found on our GitHub repo:
+https://github.com/certbot/certbot/issues?q=milestone%3A0.7.0+is%3Aissue 
+
+# 0.6.0
+## 05/12/2016 
+
+* Renamed the client from `letsencrypt` to `certbot`
```
Comment (bmw) [inline CHANGELOG.md]: Nice list. Thanks for doing this.

File: CHANGELOG.md [hunk 7]
```diff
@@ -0,0 +1,331 @@
...
+`letsencrypt-auto` or if `-n/--non-interactive/--noninteractive` is 
+included on the command line. 
+
+More details about these changes can be found on our GitHub repo:
+https://github.com/certbot/certbot/issues?q=is%3Aissue+milestone%3A0.9.0+is%3Aclosed 
+
+# 0.8.1
+## 06/14/2016
+
+
```
Comment (bmw) [inline CHANGELOG.md]: nit: extra space

File: CHANGELOG.md [hunk 8]
```diff
@@ -0,0 +1,331 @@
...
+file](https://certbot.eff.org/docs/using.html#configuration-file) to run 
+an additional command before/after any certificate is renewed. Hooks 
+will only be run if a certificate is renewed. 
+* Recategorized `-h/--help` output to improve documentation and 
+discoverability. 
+* Busybox support in certbot-auto. 
+* Many small bug fixes. 
+
+More details about these changes can be found on our GitHub repo:
+https://github.com/certbot/certbot/issues?q=is%3Aissue+milestone%3A0.10.0 is%3Aclosed
```
Comment (bmw) [inline CHANGELOG.md]: Space in the URL breaks the link. This is my fault, not yours.

File: CHANGELOG.md [hunk 9]
```diff
@@ -249,10 +248,10 @@ https://github.com/letsencrypt/letsencrypt/issues?q=is%3Aissue+milestone%3A0.4.2
 # 0.4.1
 ## 02/29/2016
 
-* Apache parsing errors with some configurations 
-* Werkzeug dependency problems on some Red Hat systems 
-* Bootstraping failures when using letsencrypt-auto with --no-self-upgrade 
-* Problems with parsing renewal config files from private beta 
+* Fixes Apache parsing errors with some configurations
+* Fixes Werkzeug dependency problems on some Red Hat systems
+* Fixes bootstraping failures when using letsencrypt-auto with --no-self-upgrade
```
Comment (strugee) [inline CHANGELOG.md]: Typo: "bootstrapping"

**Comment by strugee** [comment]
Also relevant: http://keepachangelog.com which suggests a standard format for change logs

**Comment by SwartzCr** [comment]
Fixed the typo, still waiting for the @bmw review.
I'm happy to translate them to the style that keepachangelog suggests if @bmw thinks I should, but my understanding is that we needed this ASAP for Debian compatibility - so I'd hold off on that for now

**Comment by strugee** [comment]
:thumbsup: I can submit patches on top of this fixing the format if they'll be merged; agreed that this would be useful to go in ASAP regardless of format

**Pull request merged**